File: src/main/java/ukalus/persistence/memory/MemoryLog.java
package ukalus.persistence.memory;

import ukalus.persistence.Log;
import ukalus.persistence.Reference;

import java.io.IOException;
import java.util.Iterator;
import java.util.LinkedList;
import java.util.List;
import java.util.function.Function;

public class MemoryLog implements Log {

  private List<Function> storage = new LinkedList<Function>();

  public void clear() throws IOException {
    storage.clear();
  }

  public void add(Function o) throws IOException {
    storage.add(o);
  }

  public void close() throws IOException {
  }

  public Iterator<Function> iterator() {
    return storage.iterator();
  }

}

File: src/main/java/ukalus/persistence/memory/MemoryLog.kt
package ukalus.persistence.memory

import ukalus.persistence.Log

import java.io.IOException
import java.util.LinkedList

class MemoryLog<T> : Log<T> {

    private val storage = LinkedList<T>()

    override fun clear() {
        storage.clear()
    }

    override fun add(o: T) {
        storage.add(o)
    }

    override fun close() {
    }

    override fun iterator(): Iterator<T> {
        return storage.iterator()
    }

}
